roles/pulp_database_config/tasks/main.yml

```diff
@@ -20,23 +20,28 @@
 - meta: flush_handlers
 
 - block:
-    - name: Run database auth migrations
-      command: '{{ pulp_django_admin_path }} migrate auth --no-input'
-      register: migrate_auth
-      changed_when: "'No migrations to apply' not in migrate_auth.stdout"
-      environment:
-        # Stops django-guardian from creating an anonymous user
-        PULP_ANONYMOUS_USER_NAME: '@none None'
-
     - name: Run database migrations
       command: '{{ pulp_django_admin_path }} migrate --no-input'
       register: result
       changed_when: "'No migrations to apply' not in result.stdout"
+      environment:
+        # Stops django-guardian from creating an anonymous user
+        PULP_ANONYMOUS_USER_NAME: '@none None'
+
+    - name: Check if admin password has been set
+      stat:
+        path: "{{ pulp_user_home }}/admin_password_is_set"
+        get_attributes: false
+        get_checksum: false
+        get_mime: false
+      register: admin_password_is_set
 
     - name: Set the Pulp admin user's password
-      command: '{{ pulp_django_admin_path }} reset-admin-password --password {{ pulp_default_admin_password }}'
+      shell: |
+        {{ pulp_django_admin_path }} reset-admin-password --password {{ pulp_default_admin_password }} && \
+        touch {{ pulp_user_home }}/admin_password_is_set
       no_log: true
-      when: pulp_default_admin_password is defined and migrate_auth.changed
+      when: not ( admin_password_is_set.stat.exists | default(true))
```

Comment:
`default(true)` is to prevent users that use `--start-at-task` from accidentally resetting.

Reply:
Nice! I didn't know about --start-at-task

Reply:
The problem of "--start-at-task" is that almost every use of "set_fact" renders it unusable (I believe that is the main reason to avoid them.).